### PR TITLE
Accessibility: fix speech for keysig chooser in the New Score Wizard

### PR DESCRIPTION
**Pull request by shoogle**
Resolves: https://musescore.org/en/node/301789

Fix speech for the key signature chooser in the New Score Wizard. The old chooser was a custom widget and was ignored by screen readers. The new chooser is a standard QListView, which is accessible by default. It uses the same model and icon drawing code as the QML palettes.

While implementing the new key signature chooser, it was neccesary to fix some bugs in the icon drawing code. The old code had been copied and pasted from the old (non-QML) palette, which was notoriously hard to follow. The easiest way to fix it was simply to write it again from scratch.

- [x] I signed [CLA](https://musescore.org/en/cla)
- [x] I made sure the code in the PR follows [the coding rules](https://musescore.org/en/handbook/developers-handbook/finding-your-way-around/musescore-coding-rules)
- [x] I made sure the code compiles on my machine
- [x] I made sure there are no unnecessary changes in the code
- [x] I made sure the title of the PR reflects the core meaning of the issue you are solving
- [x] I made sure the commit message(s) contain a description and answer the question "Why do those changes fix that particular issue?" or "Why are those changes really necessary as improvements?"
- [x] I made sure the commit message title starts with "fix #424242:" if there is a related issue
- [N/A] I created the test (mtest, vtest, script test) to verify the changes I made

### CODE REVIEW

File: mscore/palette/palettelistview.cpp [hunk 1]
```diff
@@ -0,0 +1,108 @@
+//=============================================================================
+//  PaletteListView
+//
+//  Copyright (C) 2020 Peter Jonas
```
Comment (Jojo-Schmitz) [inline mscore/palette/palettelistview.cpp]: Not MuseScore BVBA?

File: mscore/palette/palettelistview.h [hunk 2]
```diff
@@ -0,0 +1,60 @@
+//=============================================================================
+//  PaletteListView
+//
+//  Copyright (C) 2020 Peter Jonas
```
Comment (Jojo-Schmitz) [inline mscore/palette/palettelistview.h]: dito

File: mscore/palette/palettelistview.cpp (hunk 1 above)
Comment (shoogle) [reply mscore/palette/palettelistview.cpp]: Nope. I wrote the code so it is my copyright. MuseScore has permission to use it under the terms of the CLA, and everybody else under GPL version 2.

File: mscore/palette/palettelistview.h (hunk 2 above)
Comment (shoogle) [reply mscore/palette/palettelistview.h]: See [above](https://github.com/musescore/MuseScore/pull/5729#discussion_r381845087).

File: mscore/palette/palettelistview.cpp (hunk 1 above)
Comment (Jojo-Schmitz) [reply mscore/palette/palettelistview.cpp]: OK. Or soon under GPL version 3 I guess

Comment (shoogle) [reply mscore/palette/palettelistview.cpp]: Right, but they will always be able to use this particular file under GPL 2 if they prefer. The CLA gives MuseScore the right to offer the file under other licenses *in addition to GPL 2*. The project as a whole may become GPL 3, but this file would be dual-licensed GPL 2 + GPL 3.

File: mscore/palette/palettelistview.cpp [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+      const int nextRow = (currentRow() == count() - 1) ? 0 : currentRow() + 1;
+      const QModelIndex nextIndex = model()->index(nextRow, 0, rootIndex());
+      const auto matchedIndexList = model()->match(nextIndex, Qt::ToolTipRole, str);
+      if (!matchedIndexList.isEmpty())
+            setCurrentIndex(matchedIndexList.first());
+      }
+
+//---------------------------------------------------------
+//   onlyContainsVisibleCharacters
+// Return true iff string is non-empty and
```
Comment (dmitrio95) [inline mscore/palette/palettelistview.cpp]: Could you please make these comments (except for the function name itself) start from three slashes (`///`) to make them visible by Doxygen? This isn't an "official" convention right now but still should be useful, I believe.

Also there is a typo in `if` in this line.

Comment (shoogle) [reply mscore/palette/palettelistview.cpp]: @dmitrio95, sure. BTW, `iff` is not a typo. It is short for ["if and only if"](https://en.wikipedia.org/wiki/If_and_only_if).

Comment (Jojo-Schmitz) [reply mscore/palette/palettelistview.cpp]: better spell that out, it is not a widely known acronym

Comment (shoogle) [reply mscore/palette/palettelistview.cpp]: @dmitrio95, done!
